inference-engine/include/ie_parameter.hpp

```diff
@@ -318,6 +353,20 @@ class INFERENCE_ENGINE_API_CLASS(Parameter) {
         bool operator==(const Any& rhs) const override {
             return rhs.is(typeid(T)) && equal<T>(*this, rhs);
         }
+
+        template <class U>
+        typename std::enable_if<!HasOutputStreamOperator<U>::value, void>::type
+        print(std::ostream& stream, const U& object) const {}
```

Comment:
Consider don't mix return type with templates/concepts:
```suggestion
        template <class U,  
                  typename std::enable_if<!HasOutputStreamOperator<U>::value, bool>::type = true>
        void print(std::ostream& stream, const U& object) const {}
```

In our src base `int` is sometimes used:
```suggestion
        template <class U,  
                  typename std::enable_if<!HasOutputStreamOperator<U>::value, int>::type = 0>
        void print(std::ostream& stream, const U& object) const {}
```